File: maio/29/fizzBuzz.py
# coding: utf-8

def fizzbuzz(numero):
	if divide_por_3(numero) and divide_por_5(numero):
		return 'fizzbuzz'	
	elif divide_por_3(numero):
		return "fizz"
	elif divide_por_5(numero):
		return 'buzz'
	else:
		return numero


def divide_por_5(numero):
	return numero % 5 == 0		

def divide_por_3(numero):
	return numero % 3 == 0
		

File: novembro/27/fizzbuzz.rb

class FizzBuzz

    def initialize(numero)
        @numero = numero
        @resultado = self.regra_de_negocio
    end

    def regra_de_negocio
        resultado = ""

        resultado << 'fizz' if @numero % 3 == 0        
        resultado << 'buzz' if @numero.modulo(5) == 0
        
        if resultado.empty?
            resultado << @numero.to_s
        end

        resultado
    end

    def imprimir
        @resultado
    end

end
